clang/include/clang/InstallAPI/HeaderFile.h

```diff
@@ -97,6 +97,14 @@ class HeaderFile {
                                           Other.Excluded, Other.Extra,
                                           Other.Umbrella);
   }
+
+  bool operator<(const HeaderFile &Other) const {
```

Comment:
What's the reasoning for the ordering? Might be easier to understand in the future with some comments.

Reply:
I could infer, but I don't actually know @ributzka do you remember?

Reply:
Including the umbrella header first increases the likelihood of successfully parsing all the headers in the framework. Extra headers always go last.